cnd/src/btsieve/bitcoin/bitcoin_cache.rs

```diff
@@ -0,0 +1,94 @@
+use crate::btsieve::{BlockByHash, LatestBlock};
+use bitcoin::{hashes::sha256d, util::hash::BitcoinHash};
+use fmt::Debug;
+use futures::Future;
+use futures_core::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+};
+use std::{fmt, fmt::Formatter, sync::Arc};
+use tokio::sync::Mutex;
+
+pub struct BitcoinCache<T> {
+    pub inner: T,
```

Comment:
I don't mind too much - I would have been more abstract for the same reason as the `C` above, but changed it to your suggestion for now.

Reply:
Yeah, the lipp2p codebase uses inner all the time, I find it un-descriptive unless the outer thing is just a wrapper.  This is not really a wrapper because it has two fields with two distinct purposes.  Thanks for changing it.